src/main/scala/mesosphere/marathon/core/task/TaskStateOp.scala

```diff
@@ -0,0 +1,23 @@
+package mesosphere.marathon.core.task
+
+import mesosphere.marathon.core.task.bus.MarathonTaskStatus
+import mesosphere.marathon.state.Timestamp
+
+sealed trait TaskStateOp
+
+object TaskStateOp {
+  case class Launch(appVersion: Timestamp, status: Task.Status, networking: Task.Networking) extends TaskStateOp
+  // FIXME (3221): MarathonTaskStatus was introduced for the message bus – is it ok to use it here?
+  case class MesosUpdate(status: MarathonTaskStatus, now: Timestamp) extends TaskStateOp
+  case object Timeout extends TaskStateOp
+  //  Garbage?
```

Comment:
Garbage would not be a transition, I think. It would affect the state. If you want to destroy a task including the volume, you would need to mark the `LaunchOnReservation` as `destroyMe` and then transition to `Task.Garbage` when receiving the kill confirmation. Or something similar.